compiler/rustc_hir_analysis/src/astconv/generics.rs

```diff
@@ -590,8 +591,72 @@ pub(crate) fn check_generic_arg_count(
 }
 
 /// Emits an error regarding forbidden type binding associations
-pub fn prohibit_assoc_ty_binding(tcx: TyCtxt<'_>, span: Span) {
-    tcx.sess.emit_err(AssocTypeBindingNotAllowed { span });
+pub fn prohibit_assoc_ty_binding(
+    tcx: TyCtxt<'_>,
+    span: Span,
+    segment: Option<(&hir::PathSegment<'_>, Span)>,
+) {
+    let mut err = tcx.sess.create_err(AssocTypeBindingNotAllowed { span });
+    if let Some((segment, span)) = segment {
+        if segment.args().parenthesized {
```

Comment:
You can use let chains to eliminate one nesting level:

```suggestion
    if let Some((segment, span)) = segment && segment.args().parenthesized {
```